tools/topology/topology2/platform/intel/hdmi-generic.conf

```diff
@@ -121,7 +121,7 @@ Object.PCM {
 	}
 	pcm.7 {
 		name HDMI3
-		id 7
+		id $HDMI3_PCM_ID
```

Comment:
but here we are not adding the HDMI4 in a file that's supposed to be generic. So doesn't this result in HDMI4 NEVER being added?

FWIW, I would recommend not adding HDMI4, it's useless in practice and it makes our life problematic. If we can we should limit all topology2 new files to only use HDMI1..3

Reply:
> but here we are not adding the HDMI4 in a file that's supposed to be generic. So doesn't this result in HDMI4 NEVER being added?

HDMI4 pcm will be added in hdmi-4th.conf if needed.

> 
> FWIW, I would recommend not adding HDMI4, it's useless in practice and it makes our life problematic. If we can we should limit all topology2 new files to only use HDMI1..3

Even for SoundWire + HDMI topology?

Reply:
yes, this is painful so we should only allow for 3 HDMI streams. Adding @kv2019i to validate this direction.

Reply:
we can simplify for 3 HDMIs for all SoundWire topologies. It's fine if the machine driver exposes 4 and the topology uses only 3. What doesn't work is the other way around, a topology using more dailinks than what is exposed by the machine driver.